zebra-consensus/src/primitives/halo2/tests.rs

```diff
@@ -154,9 +157,10 @@ async fn verify_generated_halo2_proofs() {
         .clone()
         .iter()
         .map(|bytes| {
-            let maybe_shielded_data: Option<zebra_chain::orchard::ShieldedData> = bytes
-                .zcash_deserialize_into()
-                .expect("a valid orchard::ShieldedData instance");
+            let maybe_shielded_data: Option<zebra_chain::orchard::ShieldedData<OrchardVanilla>> =
+                bytes
+                    .zcash_deserialize_into()
+                    .expect("a valid orchard::ShieldedData instance");
```

Comment:
```rust
let shielded_data = zebra_test::vectors::ORCHARD_SHIELDED_DATA
    .clone()
    .iter()
    .map(|bytes| {
        bytes
            .zcash_deserialize_into::<ShieldedData<OrchardVanilla>>()
            .expect("a valid orchard::ShieldedData instance")
    })
    .collect();
```

Reply:
It should deserialize to `None` if there are no actions for OrchardVanilla (see comment on `ZcashDeserialize` impl about burns).